packages/opentelemetry-core/src/trace/sampler/ParentBasedSampler.ts

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Attributes,
+  Link,
+  Sampler,
+  SamplingResult,
+  SpanContext,
+  SpanKind,
+  TraceFlags,
+} from '@opentelemetry/api';
+import { AlwaysOffSampler } from './AlwaysOffSampler';
+import { AlwaysOnSampler } from './AlwaysOnSampler';
+import { globalErrorHandler } from '../../common/global-error-handler';
+
+/**
+ * A composite sampler that either respects the parent span's sampling decision
+ * or delegates to `delegateSampler` for root spans.
+ */
+export class ParentBasedSampler implements Sampler {
+  private root: Sampler;
+  private remoteParentSampled: Sampler;
+  private remoteParentNotSampled: Sampler;
+  private localParentSampled: Sampler;
+  private localParentNotSampled: Sampler;
+
+  constructor(config: ParentBasedSamplerConfig) {
+    this.root = config.root;
+
+    if (!this.root) {
+      globalErrorHandler(new Error("ParentBasedSampler must have a root sampler configured"))
```

Comment:
If I understand correctly, `globalErrorHandler` may be designed for those exceptions thrown at runtime asynchronously. The case here is a development time error and should abort the execution flow and do not proceed. A direct `throw` might be prominent and sufficient here.

Reply:
We have not validated/thrown in any other cases. If we want to start doing that, we should make a PR to validate configs and throw on invalid ones across the board.

Reply:
@dyladan I think we should have some default `root` here a "noop root" in this case, so then the line 67 will not fail with undefined object.